src/argilla/client/feedback/training/frameworks/sentence_transformers.py

```diff
@@ -0,0 +1,210 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from argilla.client.feedback.training.base import ArgillaTrainerSkeleton
+from argilla.client.feedback.training.schemas import TrainingTaskForSentenceSimilarity
+from argilla.training.utils import filter_allowed_args, get_default_args
+from argilla.utils.dependency import require_version
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+
+
+class ArgillaSentenceTransformersTrainer(ArgillaTrainerSkeleton):
+    _logger = logging.getLogger("ArgillaSentenceTransformersTrainer")
+    _logger.setLevel(logging.INFO)
+
+    # TODO: Update with https://github.com/argilla-io/argilla/pull/3555
+    require_version("sentence-transformers")
+
+    def __init__(
+        self,
+        dataset: "FeedbackDataset",
+        task: TrainingTaskForSentenceSimilarity,
+        prepared_data=None,
+        model: str = None,
+        seed: int = None,
+        train_size: Optional[float] = 1,
+        cross_encoder: bool = False,
+    ) -> None:
+        super().__init__(
+            dataset=dataset, task=task, prepared_data=prepared_data, model=model, train_size=train_size, seed=seed
+        )
+
+        # The prepared_data is lost in the TrainerSkeleton, is this intended?
+        self._prepared_data = prepared_data
+        self._cross_encoder = cross_encoder
+
+        if model is None:
+            if not self._cross_encoder:
```

Comment:
Perhaps we could add a warning here about the usage of a pre-defined model

Reply:
Does something like this work?:

```python
self._logger.warning(f"No model was selected, using pre-defined `{'Cross-Encoder' if self._cross_encoder else 'Bi-Encoder'}` with `{self._model}`.")
```